test/functional/test_download_and_import.py

```diff
@@ -12,45 +12,106 @@
 # language governing permissions and limitations under the License.
 from __future__ import absolute_import
 
-from sagemaker_containers import modules
+import shlex
+import subprocess
+import textwrap
+
+import pytest
+
+from sagemaker_containers import errors, modules
 import test
 
 content = ['from distutils.core import setup\n',
-           'setup(name="test_script", py_modules=["test_script"])']
+           'setup(name="my_test_script", py_modules=["my_test_script"])']
 
 SETUP = test.File('setup.py', content)
 
-USER_SCRIPT = test.File('test_script.py', 'def validate(): return True')
+USER_SCRIPT = test.File('my_test_script.py', 'def validate(): return True')
+
 
+@pytest.fixture(name='user_module_name')
+def erase_user_module():
+    user_module = 'my_test_script'
+    yield user_module
+    try:
+        subprocess.check_call(shlex.split('pip uninstall -y --quiet %s' % user_module))
+    except subprocess.CalledProcessError:
+        pass
 
-def test_download_and_import_module():
+
+def test_download_and_import_module(user_module_name):
     user_module = test.UserModule(USER_SCRIPT).add_file(SETUP).upload()
 
-    module = modules.download_and_import(user_module.url, 'test_script')
+    module = modules.download_and_import(user_module.url, user_module_name, cache=False)
 
     assert module.validate()
 
 
-def test_download_and_import_script():
+def test_download_and_import_script(user_module_name):
     user_module = test.UserModule(USER_SCRIPT).upload()
 
-    module = modules.download_and_import(user_module.url, 'test_script')
+    module = modules.download_and_import(user_module.url, user_module_name, cache=False)
 
     assert module.validate()
 
 
-content = ['import os',
-           'def validate():',
-           '    return os.path.exist("requirements.txt")']
+content = textwrap.dedent("""
+            from pyfiglet import Figlet
+
+            def say():
+                return Figlet().renderText('SageMaker').strip()
+
+""")
 
-USER_SCRIPT_WITH_REQUIREMENTS = test.File('test_script.py', content)
+USER_SCRIPT_WITH_REQUIREMENTS = test.File('my_test_script.py', content)
 
-REQUIREMENTS_FILE = test.File('requirements.txt', ['keras', 'h5py'])
+REQUIREMENTS_FILE = test.File('requirements.txt', 'pyfiglet')
 
 
-def test_download_and_import_script_with_requirements():
+def test_download_and_import_script_with_requirements(user_module_name):
     user_module = test.UserModule(USER_SCRIPT_WITH_REQUIREMENTS).add_file(REQUIREMENTS_FILE).upload()
 
-    module = modules.download_and_import(user_module.url, 'test_script')
+    module = modules.download_and_import(user_module.url, user_module_name, cache=False)
+
+    # flake8: noqa ignore=W291 trailing whitespace
```

Comment:
why do you need trailing whitespace here? does this mean things will break if someone's editor automatically deletes all trailing whitespace?

Reply:
Good call. I will remove the noqa